src/background/adblocker.js

```diff
@@ -98,25 +105,26 @@ async function updateEngines() {
       await Promise.all(
         enabledEngines
           .filter((id) => id !== engines.CUSTOM_ENGINE)
-          .map((id) =>
-            engines.update(id).then(
-              (v) => {
-                updated = updated || v;
-              },
-              () => {},
-            ),
-          ),
+          .map(async (id) => {
+            try {
+              await engines.init(id);
+              const updatedEngine = await engines.update(id);
+              updated ||= updatedEngine;
+            } catch {
+              // Ignore errors
```

Comment:
Could we log the error? Looks relevant if it fails.

Reply:
It would be redundant, as the update function already logs errors, but also rethrows, so we have to catch it here. 

The logic here did not change, I only added engine.init(), and it was more convenient to change structure to async. Before we had a promise with an empty catch function.